File: mycollab-services/src/main/java/com/mycollab/module/file/StorageUtils.java
package com.mycollab.module.file;

import com.mycollab.core.utils.StringUtils;
import com.mycollab.module.file.service.AbstractStorageService;
import com.mycollab.spring.AppContextUtil;

/**
 * @author MyCollab Ltd
 * @since 5.5.0
 */
public class StorageUtils {
    public static String getResourcePath(String documentPath) {
        return AppContextUtil.getSpringBean(AbstractStorageService.class).getResourcePath(documentPath);
    }

    public static String getLogoPath(Integer accountId, String logoName, Integer size) {
        return AppContextUtil.getSpringBean(AbstractStorageService.class).getLogoPath(accountId, logoName, size);
    }

    public static String getEntityLogoPath(Integer accountId, String id, Integer size) {
        return AppContextUtil.getSpringBean(AbstractStorageService.class).getEntityLogoPath(accountId, id, size);
    }

    public static String getFavIconPath(Integer sAccountId, String favIconName) {
        return AppContextUtil.getSpringBean(AbstractStorageService.class).getFavIconPath(sAccountId, favIconName);
    }

    public static String getAvatarPath(String userAvatarId, Integer size) {
        return AppContextUtil.getSpringBean(AbstractStorageService.class).getAvatarPath(userAvatarId, size);
    }

    public static String generateAssetRelativeLink(String resourceId) {
        return AppContextUtil.getSpringBean(AbstractStorageService.class).generateAssetRelativeLink(resourceId);
    }
}

File: mycollab-services/src/main/java/com/mycollab/module/file/StorageUtils.kt
package com.mycollab.module.file

import com.mycollab.core.utils.StringUtils
import com.mycollab.module.file.service.AbstractStorageService
import com.mycollab.spring.AppContextUtil

/**
 * @author MyCollab Ltd
 * @since 5.5.0
 */
object StorageUtils {
    @JvmStatic fun getResourcePath(documentPath: String): String {
        return AppContextUtil.getSpringBean(AbstractStorageService::class.java).getResourcePath(documentPath)
    }

    @JvmStatic fun getLogoPath(accountId: Int?, logoName: String, size: Int?): String {
        return AppContextUtil.getSpringBean(AbstractStorageService::class.java).getLogoPath(accountId, logoName, size)
    }

    @JvmStatic fun getEntityLogoPath(accountId: Int?, id: String, size: Int?): String {
        return AppContextUtil.getSpringBean(AbstractStorageService::class.java).getEntityLogoPath(accountId, id, size)
    }

    @JvmStatic fun getFavIconPath(sAccountId: Int?, favIconName: String): String {
        return AppContextUtil.getSpringBean(AbstractStorageService::class.java).getFavIconPath(sAccountId, favIconName)
    }

    @JvmStatic fun getAvatarPath(userAvatarId: String, size: Int?): String {
        return AppContextUtil.getSpringBean(AbstractStorageService::class.java).getAvatarPath(userAvatarId, size)
    }

    @JvmStatic fun generateAssetRelativeLink(resourceId: String): String {
        return AppContextUtil.getSpringBean(AbstractStorageService::class.java).generateAssetRelativeLink(resourceId)
    }
}
